### PR TITLE
[WIP] Use struct for colors

### PR DESCRIPTION
**Pull request by 7415963987456321**
Instead of having 3 different variables, put all the colors in a struct. Just wanted to get your input before I go further with this. I'm thinking that more could be done with change, i.e. maybe put the old_background_color inside the struct as well and maybe change the logic for that....  
Also I have yet to fix the indentation.

### CODE REVIEW

**Comment by francma** [comment]
I like this :+1: 

> Maybe put the old_background_color inside the struct as well

Yes, that would be nice.

You can also drop `color` postfix from struct members as it is already clear from struct name that it is color.

> Also I have yet to fix the indentation.

Check out clang-format - in most distributions it's part of a clang package. To fix indentation & coding styles you just  run `clang-format -i [file_name]` and that's it.

**Comment by 7415963987456321** [comment]
All comments have been adressed, removed the postfix and fixed the indent.

**Comment by francma** [comment]
I think we misunderstood each other. By: 

>  Maybe put the old_background_color inside the struct as well

I imagined having struct like this:

```

**Comment by 7415963987456321** [comment]
Ah so you mean to have another instance of the same struct for the old variables, that makes sense.

File: wob.c [hunk 1]
```diff
@@ -654,9 +663,9 @@ main(int argc, char **argv)
 					return EXIT_SUCCESS;
 				}
 
-				old_background_color = background_color;
-				old_border_color = border_color;
-				if (fgets_rv == NULL || !wob_parse_input(input_buffer, &percentage, &background_color, &border_color, &bar_color) || percentage > maximum) {
+				old_colors.background = colors.background;
```
Comment (francma) [inline wob.c]: I think this can be simplified to just:
```suggestion
				old_colors = colors;
```

File: wob.c [hunk 2]
```diff
@@ -598,13 +604,17 @@ main(int argc, char **argv)
...
-	argb_color background_color = BLACK;
-	argb_color bar_color = WHITE;
-	argb_color border_color = WHITE;
+	struct wob_colors colors = {
+		.background = BLACK,
+		.bar = WHITE,
+		.border = WHITE,
+	};
+
+	struct wob_colors old_colors = {0, 0, 0};
```
Comment (francma) [inline wob.c]: This is equivalent to above:

```suggestion
	struct wob_colors old_colors = {0};
```

File: wob.c (hunk 1 above)
Comment (7415963987456321) [reply wob.c]: Of course.

File: wob.c (hunk 2 above)
Comment (7415963987456321) [reply wob.c]: fixed.

**Comment by 7415963987456321** [comment]
All comments adressed :)

File: wob.c [hunk 3]
```diff
@@ -654,9 +663,8 @@ main(int argc, char **argv)
 					return EXIT_SUCCESS;
 				}
 
-				old_background_color = background_color;
-				old_border_color = border_color;
-				if (fgets_rv == NULL || !wob_parse_input(input_buffer, &percentage, &background_color, &border_color, &bar_color) || percentage > maximum) {
+                old_colors = colors;
```
Comment (francma) [inline wob.c]: Tabs please :slightly_smiling_face:

Comment (7415963987456321) [reply wob.c]: Ah, sorry about that dude totally forgot to run clang-format  :man_facepalming: . Should all be fixed now

**Comment by francma** [comment]
Thanks :slightly_smiling_face:

**Pull request merged**